views/templates/hook/ps_emailsubscription.tpl

```diff
@@ -30,6 +30,12 @@
   {/if}
   <form action="{$urls.pages.index}" method="post">
     <input type="text" name="email" value="{$value}" placeholder="{l s='Your e-mail' mod='ps_emailsubscription'}" />
+    {if $need_confirmation}
+    <span class="custom-checkbox">
+       <input type="checkbox" name="confirm-optin" value="1" required>
+       <label>{l s='I agree to receive newsletter emails and I am aware of [1]the privacy policy[/1]' tags=['<a href="%s">'|sprintf:$link->getCMSLink($cms_page)] mod='ps_emailsubscription'}</label>
```

Comment:
Please, no more call to a function like that in the them. Moreover, the object should no not be passed to the template anymore.
Can you generate the url in the module and pass it to the template, please? Also, think to update the Classic theme 😄 

Thx